packages/webpack-module-federation/src/loadRemote.ts

```diff
@@ -1,7 +1,17 @@
-import { isNil } from "@squide/core";
+import { type ModuleRegisterFunction, isNil } from "@squide/core";
+
+// The interface of a module that can be loaded dynamically.
+interface Module {
+    register?: ModuleRegisterFunction;
+}
 
 // Webpack globals we need to access when loading Federated Modules dynamically
 // See: https://webpack.js.org/concepts/module-federation/#dynamic-remote-containers.
+interface Container {
+    init: (shareScope: unknown) => void;
+    get: (module: string) => () => Module;
+}
```

Comment:
I don't think I would add this. This code is mostly copied from Webpack docs and this is not a piece of code that I expect us to evolve on our own. I rather copy/paste updated versions from the Webpack docs.

Reply:
I think its fair. i only kept the method's return value instead of the whole container